sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -69,17 +61,23 @@ impl Policy for BearerTokenCredentialPolicy {
         request: &mut Request,
         next: &[Arc<dyn Policy>],
     ) -> PolicyResult {
-        if self.is_token_expired() {
-            let access_token = self.credential.get_token(&self.scopes()).await?;
-            self.refresh_token(&access_token);
+        let access_token = self.access_token.as_ref().read().await;
+
+        if let Some(token) = &(*access_token) {
+            if token.is_expired(Some(DEFAULT_REFRESH_TIME)) {
+                drop(access_token);
```

Comment:
Don't call this explicitly. It will be dropped when it gets shadowed in the line below. Same with the else below. This isn't idiomatic or necessary.